flake.nix

```diff
@@ -8,13 +8,19 @@
 
   outputs = { self, nixpkgs, ... }:
     let
+      lib = nixpkgs.lib;
       supportedSystems = [
         "x86_64-linux"
         "i686-linux"
         "aarch64-linux"
         "riscv64-linux"
       ];
-      forAllSystems = nixpkgs.lib.genAttrs supportedSystems;
+      forAllSystems = lib.genAttrs supportedSystems;
+
+      versionInfo = import ./version.nix;
+      version = versionInfo.version +
+        (lib.optionalString (!versionInfo.released)
+          "-${self.shortRev or self.dirtyShortRev or "dirty"}");
```

Comment:
Can we skip the revision for now? This will trigger a rebuild on very commit no matter if anything changed.

Reply:
If we encourage users to use a release version, than error reporting should be no longer a big issue.

Reply:
Ah yes, I didn't consider that. I'd still like to have the `-dirty` suffix for all non-released versions, I think that's a reasonable compromise.